easybuild/easyconfigs/s/snakemake/snakemake-7.18.2-foss-2021b.eb

```diff
@@ -0,0 +1,96 @@
+easyblock = 'PythonBundle'
+
+name = 'snakemake'
+version = '7.18.2'
+
+homepage = 'https://snakemake.readthedocs.io'
+description = "The Snakemake workflow management system is a tool to create reproducible and scalable data analyses."
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('GitPython', '3.1.24'),
+    ('IPython', '7.26.0'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+exts_list = [
+    ('wrapt', '1.14.1', {
+        'checksums': ['380a85cf89e0e69b7cfbe2ea9f765f004ff419f34194018a6827ac0e3edfed4d'],
+    }),
+    ('ratelimiter', '1.2.0.post0', {
+        'checksums': ['5c395dcabdbbde2e5178ef3f89b568a3066454a6ddc223b76473dac22f89b4f7'],
+    }),
+    ('ConfigArgParse', '1.5.3', {
+        'checksums': ['1b0b3cbf664ab59dada57123c81eff3d9737e0d11d8cf79e3d6eb10823f1739f'],
+    }),
+    ('datrie', '0.8.2', {
+        'checksums': ['525b08f638d5cf6115df6ccd818e5a01298cd230b2dac91c8ff2e6499d18765d'],
+    }),
+    ('toposort', '1.7', {
+        'checksums': ['ddc2182c42912a440511bd7ff5d3e6a1cabc3accbc674a3258c8c41cbfbb2125'],
+    }),
+    ('amply', '0.1.5', {
+        'checksums': ['ad717b490b6b7055909faa195e82a4432b706f8f95854050172f42931e478426'],
+    }),
+    ('PuLP', '2.7.0', {
+        'checksums': ['e73ee6b32d639c9b8cf4b4aded334ba158be5f8313544e056f796ace0a10ae63'],
+    }),
+    ('smart_open', '6.2.0', {
+        'checksums': ['1b4df5c8365218f3852c507451920ccad606c80b0acb4e67508e50ba9b5d2632'],
+    }),
+    ('connection_pool', '0.0.3', {
+        'checksums': ['bf429e7aef65921c69b4ed48f3d48d3eac1383b05d2df91884705842d974d0dc'],
+    }),
+    ('stopit', '1.1.2', {
+        'checksums': ['f7f39c583fd92027bd9d06127b259aee7a5b7945c1f1fa56263811e1e766996d'],
+    }),
+    ('reretry', '0.11.1', {
+        'checksums': ['4ae1840ae9e443822bb70543c485bb9c45d1d009e32bd6809f2a9f2839149f5d'],
+    }),
+    ('throttler', '1.2.1', {
+        'checksums': ['8b23d3485a96d98484024a850c1887ccc685bead17e86c8a9e4b0335e7d74778'],
+        # workaround for https://github.com/uburuntu/throttler/issues/3
+        'preinstallopts': "touch readme.md requirements-dev.txt && ",
+    }),
+    ('dpath', '2.0.6', {
+        'checksums': ['5a1ddae52233fbc8ef81b15fb85073a81126bb43698d3f3a1b6aaf561a46cdc0'],
+    }),
+    ('plac', '1.3.5', {
+        'checksums': ['38bdd864d0450fb748193aa817b9c458a8f5319fbf97b2261151cfc0a5812090'],
+    }),
+    # yte requires PyYAML >= 6.0
```

Comment:
This is risky, loading another module combined with snakemake that depends on PyYAML v5.4.1 from this toolchain is not unlikely. In that case one might break the other depending on the load order.

Looking at the [changelog](https://github.com/yaml/pyyaml/blob/master/CHANGES) of PyYAML, one point of failure is that v6.0 made the `Loader` argument in `yaml.load()` mandatory.

I suggest to move this version of snakemake to `foss/2022a`.

Reply:
> but we take the same approach (installing a newer Python package as extension if needed) for various other tools as well

Indeed, and that's unreliable as a general approach, this is a good example for #15004

I won't block this PR though, as we don't have any clear guidelines in this regard.